core/storage/hamt/hamt.hpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_STORAGE_HAMT_HAMT_HPP
+#define CPP_FILECOIN_STORAGE_HAMT_HAMT_HPP
+
+#include <string>
+#include <vector>
+
+#include <boost/variant.hpp>
+
+#include "common/outcome_throw.hpp"
+#include "common/visitor.hpp"
+#include "primitives/big_int.hpp"
+#include "storage/ipfs/datastore.hpp"
+
+namespace fc::storage::hamt {
+  enum class HamtError { EXPECTED_CID = 1, NOT_FOUND, MAX_DEPTH };
+
+  using CID = libp2p::multi::ContentIdentifier;
+  using fc::primitives::UBigInt;
+  using Value = ipfs::IpfsDatastore::Value;
+
+  extern const CID kDummyCid;
+
+  constexpr size_t kLeafMax = 3;
+
+  /** Hamt node interface */
```

Comment:
I don't get this comment quite clearly. Why is it called an interface if it's a structure?